ci-operator/step-registry/rosa/cluster/deprovision/rosa-cluster-deprovision-commands.sh

```diff
@@ -42,17 +63,33 @@ if [[ -z "$CLUSTER_ID" ]]; then
 fi
 
 echo "Deleting cluster-id: ${CLUSTER_ID}"
+start_time=$(date +"%s")
 rosa delete cluster -c "${CLUSTER_ID}" -y
 while rosa describe cluster -c "${CLUSTER_ID}" ; do
-  sleep 60
+  current_time=$(date +"%s")
+  if (( "${current_time}" - "${start_time}" >= "${DESTROY_TIMEOUT}" )); then
+    echo "error: Cluster not deleted after ${DESTROY_TIMEOUT}"
+    exit 1
+  else
+    echo "Cluster ${CLUSTER_ID} is still alive, waiting 60 seconds for the next check"
+    sleep 60
```

Comment:
Old code have no ending condition, if there is any failure deleting the cluster, it will never end, I have used the same logic than the provision job to wait up to CLUSTER_TIMEOUT for the cluster to be installed. so here are waiting up to DESTROY_TIMEOUT to cluster to be destroyed

Reply:
if we wait for the pod timeout, we cannot manage a failure, it is better to detect that the cluster is not uninstalled during the process, so we have a change to fix/report/log that failure

Reply:
After we run `rosa delete cluster`, the cluster status will be in 'uninstalling' status, the action can not be rollback. If the cluster fails deprovision, the cluster maybe hung at uninstalling status or fails in error status. The cluster will be kept here, what we can do is debug/report the failure, we can not fix the failure. 
1. 60m is enough for deprovision.
2. There is no difference between your updating and the old code. Your changes only adds some output logs.

Reply:
while true and check the rosa cluster state inside the loop

Reply:
done